File: FitMealAI/Resources/MockData/MockData.swift
//
//  MockData.swift
//  FitMealAI
//
//  Centralized sample data for SwiftUI previews. Do NOT use in
//  production paths - replace with real services in later phases.
//

import Foundation

enum MockData {

    // MARK: - User Goal

    static let user = UserGoal(
        name: "Alex",
        goal: .loseWeight,
        diet: .noRestrictions,
        dailyCalorieTarget: 2200
    )

    // MARK: - Auth

    static let credentials = AuthCredentials(
        mode: .email,
        emailOrPhone: "alex@example.com",
        password: ""
    )

    // MARK: - Preferences

    static let workoutPrefs = WorkoutPrefs(
        types: ["strength", "hiit"],
        days: "4 days",
        duration: "45 min"
    )

    static let mealPrefs = MealPrefs(
        diets: ["balanced", "high-protein"],
        timings: ["breakfast", "lunch", "dinner"],
        cookTime: "30 min",
        allergies: ["Peanuts"]
    )

    // MARK: - Meal Plan with ingredients

    static let todayMealPlan = MealPlan(
        date: Date(),
        meals: [
            Meal(
                type: .breakfast,
                title: "Greek Yogurt Bowl",
                calories: 380,
                proteinGrams: 28,
                carbsGrams: 42,
                fatGrams: 9,
                ingredients: [
                    Ingredient(name: "Greek yogurt", grams: 200, calories: 200, proteinGrams: 20, carbsGrams: 12, fatGrams: 6),
                    Ingredient(name: "Granola",      grams: 40,  calories: 160, proteinGrams: 4,  carbsGrams: 24, fatGrams: 5),
                    Ingredient(name: "Blueberries",  grams: 60,  calories: 35,  proteinGrams: 0,  carbsGrams: 9,  fatGrams: 0),
                    Ingredient(name: "Honey",        grams: 10,  calories: 30,  proteinGrams: 0,  carbsGrams: 8,  fatGrams: 0)
                ]
            ),
            Meal(
                type: .lunch,
                title: "Grilled Chicken Quinoa",
                calories: 620,
                proteinGrams: 45,
                carbsGrams: 60,
                fatGrams: 18,
                ingredients: [
                    Ingredient(name: "Chicken breast", grams: 180, calories: 300, proteinGrams: 36, carbsGrams: 0,  fatGrams: 12),
                    Ingredient(name: "Quinoa",         grams: 120, calories: 220, proteinGrams: 8,  carbsGrams: 40, fatGrams: 4),
                    Ingredient(name: "Mixed greens",   grams: 80,  calories: 60,  proteinGrams: 1,  carbsGrams: 12, fatGrams: 0),
                    Ingredient(name: "Olive oil",      grams: 10,  calories: 90,  proteinGrams: 0,  carbsGrams: 0,  fatGrams: 10)
                ]
            ),
            Meal(
                type: .dinner,
                title: "Salmon with Veggies",
                calories: 540,
                proteinGrams: 38,
                carbsGrams: 30,
                fatGrams: 22,
                ingredients: [
                    Ingredient(name: "Salmon fillet",   grams: 160, calories: 320, proteinGrams: 32, carbsGrams: 0,  fatGrams: 18),
                    Ingredient(name: "Sweet potato",    grams: 150, calories: 130, proteinGrams: 2,  carbsGrams: 30, fatGrams: 0),
                    Ingredient(name: "Roasted broccoli", grams: 120, calories: 90, proteinGrams: 4, carbsGrams: 14, fatGrams: 4)
                ]
            ),
            Meal(
                type: .snack,
                title: "Almonds & Apple",
                calories: 220,
                proteinGrams: 6,
                carbsGrams: 24,
                fatGrams: 12,
                ingredients: [
                    Ingredient(name: "Almonds", grams: 25,  calories: 145, proteinGrams: 6, carbsGrams: 5,  fatGrams: 12),
                    Ingredient(name: "Apple",   grams: 150, calories: 75,  proteinGrams: 0, carbsGrams: 19, fatGrams: 0)
                ]
            )
        ]
    )

    // MARK: - Workout Plan

    static let todayWorkout = WorkoutPlan(
        title: "Full Body Strength",
        date: Date(),
        estimatedMinutes: 35,
        exercises: [
            Exercise(name: "Push-ups",        sets: 3, reps: 12, isCompleted: true),
            Exercise(name: "Squats",          sets: 3, reps: 15, isCompleted: true),
            Exercise(name: "Plank",           sets: 3, reps: 1, durationSeconds: 45),
            Exercise(name: "Lunges",          sets: 3, reps: 12),
            Exercise(name: "Mountain Climbers", sets: 3, reps: 20)
        ]
    )

    // MARK: - Habits

    static let habits: [Habit] = [
        Habit(title: "Drink 2L water",   iconSystemName: "drop.fill",         isCompleted: true,  streakDays: 7),
        Habit(title: "Sleep 8 hours",    iconSystemName: "moon.fill",         isCompleted: false, streakDays: 3),
        Habit(title: "10k steps",        iconSystemName: "figure.walk",       isCompleted: true,  streakDays: 12),
        Habit(title: "5 min meditation", iconSystemName: "leaf.fill",         isCompleted: false, streakDays: 0),
        Habit(title: "No sugar",         iconSystemName: "xmark.circle.fill", isCompleted: false, streakDays: 1)
    ]

    // MARK: - Subscription Plans

    static let plans: [SubscriptionPlan] = [
        SubscriptionPlan(
            tier: .free,
            pricePerMonth: "$0",
            features: ["1 AI meal plan / day", "10 catalog shuffles / day", "Daily plan + habit tracker"]
        ),
        SubscriptionPlan(
            tier: .silver,
            pricePerMonth: "$4.99",
            features: ["20 AI generations / day", "Unlimited catalog shuffles", "Weekly meal plans", "Workout variations"]
        ),
        SubscriptionPlan(
            tier: .gold,
            pricePerMonth: "$9.99",
            features: ["30 AI generations / day", "Unlimited catalog shuffles", "Advanced progress analytics", "Priority AI coaching"],
            isHighlighted: true
        )
    ]

    // MARK: - Payment Request

    static let pendingPayment = PaymentRequest(
        tier: .gold,
        amount: "$9.99",
        transactionId: "ABA-TX-00042",
        screenshotFileName: "receipt.png",
        status: .pending,
        submittedAt: Date()
    )
}

File: android/app/src/main/java/com/fitmealai/data/MockData.kt
package com.fitmealai.data

import com.fitmealai.domain.Exercise
import com.fitmealai.domain.FitnessGoal
import com.fitmealai.domain.Habit
import com.fitmealai.domain.Ingredient
import com.fitmealai.domain.Meal
import com.fitmealai.domain.MealPlan
import com.fitmealai.domain.MealType
import com.fitmealai.domain.SubscriptionPlan
import com.fitmealai.domain.SubscriptionTier
import com.fitmealai.domain.UserProfile
import com.fitmealai.domain.WorkoutPlan

object MockData {
    val user = UserProfile(
        id = "preview-user",
        name = "Alex",
        email = "alex@fitmeal.ai",
        goal = FitnessGoal.EatHealthier,
        tier = SubscriptionTier.Free,
        dailyCalorieTarget = 2_000,
    )

    val mealPlan = MealPlan(
        dateLabel = "Today",
        meals = listOf(
            Meal(
                id = "breakfast",
                type = MealType.Breakfast,
                title = "Greek yogurt bowl",
                description = "Creamy yogurt with granola, fresh berries, and honey.",
                calories = 380,
                proteinGrams = 28,
                carbsGrams = 42,
                fatGrams = 9,
                ingredients = listOf(
                    Ingredient("Greek yogurt", 200, 200, 20, 12, 6),
                    Ingredient("Granola", 40, 160, 4, 24, 5),
                    Ingredient("Blueberries", 60, 35, 0, 9, 0),
                    Ingredient("Honey", 10, 30, 0, 8, 0),
                ),
                recipeSteps = listOf(
                    "Spoon yogurt into a bowl.",
                    "Top with granola and blueberries.",
                    "Drizzle honey on top.",
                ),
            ),
            Meal(
                id = "lunch",
                type = MealType.Lunch,
                title = "Grilled chicken quinoa bowl",
                description = "High-protein lunch with mixed greens and olive oil.",
                calories = 620,
                proteinGrams = 45,
                carbsGrams = 60,
                fatGrams = 18,
                ingredients = listOf(
                    Ingredient("Chicken breast", 180, 300, 36, 0, 12),
                    Ingredient("Quinoa", 120, 220, 8, 40, 4),
                    Ingredient("Mixed greens", 80, 60, 1, 12, 0),
                    Ingredient("Olive oil", 10, 90, 0, 0, 10),
                ),
                recipeSteps = listOf(
                    "Cook quinoa per package directions.",
                    "Season and grill chicken until 74C internal.",
                    "Plate greens, top with chicken and quinoa, finish with olive oil.",
                ),
            ),
            Meal(
                id = "dinner",
                type = MealType.Dinner,
                title = "Salmon with roasted veggies",
                description = "Omega-rich salmon with sweet potato and broccoli.",
                calories = 540,
                proteinGrams = 38,
                carbsGrams = 30,
                fatGrams = 22,
                ingredients = listOf(
                    Ingredient("Salmon fillet", 160, 320, 32, 0, 18),
                    Ingredient("Sweet potato", 150, 130, 2, 30, 0),
                    Ingredient("Roasted broccoli", 120, 90, 4, 14, 4),
                ),
                recipeSteps = listOf(
                    "Roast sweet potato and broccoli at 220C for 20 minutes.",
                    "Pan-sear salmon skin-side down 4 minutes, flip, 2 more.",
                    "Plate together.",
                ),
            ),
        ),
    )

    val workout = WorkoutPlan(
        title = "Upper body strength",
        estimatedMinutes = 45,
        exercises = listOf(
            Exercise("ex-1", "Push ups", sets = 3, reps = 12, restSeconds = 60),
            Exercise("ex-2", "Dumbbell rows", sets = 3, reps = 10, restSeconds = 75),
            Exercise("ex-3", "Shoulder press", sets = 3, reps = 10, restSeconds = 75),
            Exercise("ex-4", "Plank", sets = 3, reps = 1, restSeconds = 45, notes = "Hold 45s"),
        ),
    )

    val habits = listOf(
        Habit("h-1", "Drink 8 glasses of water", emoji = "💧", streakDays = 12, isCompleted = true),
        Habit("h-2", "Walk 8,000 steps", emoji = "🚶", streakDays = 6, isCompleted = false),
        Habit("h-3", "Sleep before 11 PM", emoji = "😴", streakDays = 9, isCompleted = false),
        Habit("h-4", "Stretch for 5 minutes", emoji = "🧘", streakDays = 4, isCompleted = true),
    )

    val plans = listOf(
        SubscriptionPlan(
            tier = SubscriptionTier.Free,
            priceLabel = "$0",
            tagline = "The basics, free forever",
            perks = listOf(
                "1 AI meal plan / day",
                "10 catalog shuffles / day",
                "Today's plan only",
                "Habit tracker",
            ),
        ),
        SubscriptionPlan(
            tier = SubscriptionTier.Silver,
            priceLabel = "$4.99 / mo",
            tagline = "Plenty of fresh plans every day",
            perks = listOf(
                "20 AI generations / day",
                "Unlimited catalog shuffles",
                "Tomorrow's plan",
                "Workout timer",
                "Saved preferences",
            ),
            highlight = true,
        ),
        SubscriptionPlan(
            tier = SubscriptionTier.Gold,
            priceLabel = "$9.99 / mo",
            tagline = "Coach-grade plans",
            perks = listOf(
                "30 AI generations / day",
                "Unlimited catalog shuffles",
                "Weekly meal plans",
                "Advanced progress charts",
                "Priority regeneration",
                "Image cache for every meal",
            ),
        ),
    )

    val mealCookTimes = listOf("15 min", "30 min", "45 min", "60 min")
    val workoutDayChoices = listOf("2-3 days", "3-4 days", "5-6 days", "Daily")
    val workoutDurationChoices = listOf("15-30 min", "30-45 min", "45-60 min", "60+ min")
}
